tests/test_permission.py

```diff
@@ -0,0 +1,251 @@
+                update_time="2000-01-01T01:01:01.123456Z",
+            )
+
+        @add_client_method
+        def create_permission(
+            request: glm.CreatePermissionRequest,
+        ) -> glm.Permission:
+            self.observed_requests.append(request)
+            return glm.Permission(
+                name="corpora/demo_corpus/permissions/123456789",
+                role=2,
+                grantee_type=3,
+            )
+
+        @add_client_method
+        def delete_permission(
+            request: glm.DeletePermissionRequest,
+        ) -> None:
+            self.observed_requests.append(request)
+            return None
+
+        @add_client_method
+        def get_permission(
+            request: glm.GetPermissionRequest,
+        ) -> glm.Permission:
+            self.observed_requests.append(request)
+            return glm.Permission(
+                name="corpora/demo_corpus/permissions/123456789",
+                role=2,
+                grantee_type=3,
+            )
+
+        @add_client_method
+        def list_permissions(
+            request: glm.ListPermissionsRequest,
+        ) -> glm.ListPermissionsResponse:
+            self.observed_requests.append(request)
+            return [
+                glm.Permission(
+                    name="corpora/demo_corpus/permissions/123456789",
+                    role=2,
+                    grantee_type=3,
+                ),
+                glm.Permission(
+                    name="corpora/demo_corpus/permissions/987654321",
+                    role=3,
+                    grantee_type=3,
+                    email_address="_",
+                ),
+            ]
+
+        @add_client_method
+        def update_permission(
+            request: glm.UpdatePermissionRequest,
+        ) -> glm.Permission:
+            self.observed_requests.append(request)
+            return glm.Permission(
+                name="corpora/demo_corpus/permissions/123456789",
+                role=3,
+                grantee_type=3,
+            )
+
+    @parameterized.named_parameters(
+        [
+            dict(
+                testcase_name="create_permission_success",
+                role=2,
+                grantee_type=3,
+                email_address=None,
+            ),
+            dict(
+                testcase_name="create_permission_failure_email_set_when_grantee_type_is_everyone",
+                role=2,
+                grantee_type=3,
+                email_address="_",
+            ),
+            dict(
+                testcase_name="create_permission_failure_email_not_set_when_grantee_type_is_not_everyone",
+                role=2,
+                grantee_type=1,
+                email_address=None,
+            ),
+        ]
+    )
+    def test_create_permission(self, role, grantee_type, email_address):
+        x = retriever.create_corpus("demo-corpus")
+        if (role, grantee_type, email_address) == (2, 3, None):
```

Comment:
Using a switch to split the code paths in a parameterized defeats the purpose of parameterized, and makes things less direct. 

Just define three separate tests.